api/config.js

```diff
@@ -0,0 +1,63 @@
+import {createRemoteJWKSet, jwtVerify} from "jose";
+import request from "needle";
+
+const {AUTH_SERVER_URL, AUTH_REALM, GATEWAY_ACCOUNT_URL, GATEWAY_KEY} = process.env;
+// Credit: https://github.com/keycloak/keycloak-nodejs-connect/issues/492#issuecomment-1603213677
+const jwks = createRemoteJWKSet(new URL(
+	`${AUTH_SERVER_URL}/realms/${AUTH_REALM}/protocol/openid-connect/certs`));
+const GATEWAY = new class {
+	accountUrl = new URL(GATEWAY_ACCOUNT_URL);
+	accountName = this.accountUrl.pathname.slice(1);
+};
+
+// noinspection JSUnusedGlobalSymbols
+/**
+ * @param {import('@vercel/node').VercelRequest} req
+ * @param {import('@vercel/node').VercelResponse} res
+ */
+export default async function (req, res) {
+	const sendErr = (code, text, error) => {
+		console.warn(error);
+		res.status(code).json({text, error});
+	};
+	const throwErr = (code, text) => error => {
+		throw {send: () => sendErr(code, text, error)};
+	};
+	try {
+		const subdomain = req.query.domain;
+		const key = req.body;
+		const [type, token] = req.headers.authorization?.split(" ") ?? [];
+		if (type !== "Bearer")
+			return sendErr(401, 'Unauthorised');
+		const {payload} = await jwtVerify(token, jwks)
+			.catch(throwErr(401, 'Unauthorised'));
+		await checkAccess(payload.sub, subdomain)
+			.catch(throwErr(403, 'Forbidden'));
+		const pid = `${AUTH_SERVER_URL}/users/${payload.sub}`;
+		const configRes = await request('put', new URL(
+			`/api/v1/domain/${GATEWAY.accountName}/${subdomain}`, GATEWAY.accountUrl
+		).toString(), {
+			useSignatures: 'public' in key,
+			user: {'@id': pid, key}
+		}, {
+			json: true,
+			auth: 'basic',
+			username: GATEWAY.accountName,
+			password: GATEWAY_KEY
+		});
+		if (configRes.statusCode !== 200)
+			return sendErr(configRes.statusCode, configRes.statusMessage, configRes.body);
+		res.status(200).json({config: configRes.body, pid});
```

Comment:
Once the user's IdP token is validated and user access is checked, the lambda calls the Gateway to obtain the subdomain configuration, which will include a newly-minted JWT for the client. If a public key has been passed as the request body, it is forwarded to the Gateway, which will enter the public key into the subdomain for signature verifications.